controllers/storagecluster/storagecluster_controller.go

```diff
@@ -166,13 +166,17 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
-	onboardingSecretPredicates := builder.WithPredicates(
-		predicate.NewPredicateFuncs(
-			func(client client.Object) bool {
-				return client.GetName() == onboardingTicketPublicKeySecretName
-			},
-		),
-	)
+	secretPredicates := predicate.Funcs{
```

Comment:
No reason to have a struct that is not wrapped in a predicate. 

```suggestion
	secretPredicates := builder. WithPredicates(predicate.Funcs{
```